common/src/txn.ts

```diff
@@ -20,7 +20,8 @@ type AnyTxnType =
   | QfPayment
   | QfAddPool
   | QfDividend
-type SourceType = 'USER' | 'CONTRACT' | 'CHARITY' | 'BANK'
+  | AdRedeem
+type SourceType = 'USER' | 'CONTRACT' | 'CHARITY' | 'BANK' | 'AD'
```

Comment:
The destination could be 'AD' as well if there's a txn for depositing into the ad.